electron-builder.js

```diff
@@ -21,18 +24,60 @@ const fs = require("fs");
  */
 
 const NIGHTLY_APP_ID = "im.riot.nightly";
-const NIGHTLY_APP_NAME = "element-desktop-nightly";
 const NIGHTLY_DEB_NAME = "element-nightly";
 
 const pkg = JSON.parse(fs.readFileSync("package.json", "utf8"));
 
+let buildMacOsUniversal = false;
+
 /**
  * @type {import('electron-builder').Configuration}
  * @see https://www.electron.build/configuration/configuration
  */
 const config = {
     appId: "im.riot.app",
     asarUnpack: "**/*.node",
+    beforePack: async (context) => {
+        if (context.electronPlatformName === "darwin" && context.arch === Arch.universal) {
+            buildMacOsUniversal = true;
+        }
+    },
+    afterPack: async (context) => {
+        if (context.electronPlatformName !== "darwin" || context.arch === Arch.universal || !buildMacOsUniversal) {
```

Comment:
this logic, combined with that in `beforePack`, is breaking my brain. Isn't this condition always true?
 * if `electronPlatformName !== "darwin"`, it's trivially true
 * otherwise, if `electronPlatformName === "darwin"`:
    * if `context.arch === Arch.universal`, it's true
    * otherwise,  `electronPlatformName === "darwin"` and `context.arch !== Arch.universal`, so `buildMacOsUniversal` is `false`, so again it's true

Come to that, what is even the point of the `buildMacOsUniversal` flag if we have to redo most of the logic?

Reply:
> Come to that, what is even the point of the buildMacOsUniversal flag if we have to redo most of the logic?

beforePack and afterPack are called multiple times on any one run, `buildMacOsUniversal` lets us figure out whether this run includes a MacOS Universal package or not. The way the universal package works is it is a package of the arm64 and amd64 packages, so we only need to burn fuses on the final package, if universal is enabled then there's 1 final package, otherwise the constituent packages would be the final packages (arm64&/amd64)

> otherwise, electronPlatformName === "darwin" and context.arch !== Arch.universal, so buildMacOsUniversal is false, so again it's true

This is only the case in isolation, if beforePack and afterPack is called

---

Turns out though that beforePack is never called for a Universal build, only afterPack, so `buildMacOsUniversal` is never set and I'm just not going to handle the non-universal case at this time as there's no easy way for us to detect whether Universal is enabled or not via the electron-builder command called